src/alchemlyb/convergence/convergence.py

```diff
@@ -216,16 +223,25 @@ def fwdrev_cumavg_Rc(series, precision=0.01, tol=2):
             8  3.042032  3.046597            0.9
             9  3.044149  3.044385            1.0
 
-    Note
-    ----
-    This function computes :math:`R_c` from equation 16 from
-    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8397498/#FD16. The code is
-    modified based on Shujie Fan's (@VOD555) work.
-   
-   Please cite [Fan2021]_ when using this function.
-   
+    Notes
+    -----
+    This function computes :math:`R_c` from `equation 16`_ from [Fan2021]_.
+    The code is modified based on Shujie Fan's (@VOD555) work. Zhiyi Wu
+    (@xiki-tempula) improved the performance of the original algorithm by
+    pre-computing block averages for fractional increments of `precision`.
```

Comment:
Shujie Fan's (@VOD555) work has already implemented the pre-computing block averages. His work uses the normal python for loop and I have improved the performance by vectorise it using numpy array.

Reply:
thanks, reworded